app/models/user.rb

```diff
@@ -0,0 +1,9 @@
+class User < ApplicationRecord
+  befor_email{email.downcase!}
```

Comment:
`email.downcase!` chị sử thành `email.downcase` có được khoong? giải thích tại sao

Reply:
Dạ nếu chị sửa thành email.downcase thì chị phải viết lại là self.email = email. downcase. Bởi vì chúng ta đang cố gắng đưa gmail về toàn bộ chữ thường để kiểm tra validate, nếu chị chỉ bỏ dấu ! ra thôi thì sau câu lệnh đó, email nó sẽ trở về ban đầu("chưa về toàn bộ chữ thương ") thì coi như ta viết lệnh đó như vô ích ạ